docs/conf.py

```diff
@@ -0,0 +1,275 @@
+# documents.
+# Ref: python-attrs/attrs#571
+default_role = 'any'
+
+# If true, '()' will be appended to :func: etc. cross-reference text.
+add_function_parentheses = True
+
+# If true, the current module name will be prepended to all description
+# unit titles (such as .. function::).
+add_module_names = True
+
+# If true, sectionauthor and moduleauthor directives will be shown in the
+# output. They are ignored by default.
+show_authors = True
+
+# The name of the Pygments (syntax highlighting) style to use.
+# pygments_style = 'sphinx'
+
+# Add any Sphinx extension module names here, as strings. They can be
+# extensions coming with Sphinx (named 'sphinx.ext.*') or your custom
+# ones.
+extensions = [
+    # stdlib-party extensions:
+    'sphinx.ext.autodoc',
+    'sphinx.ext.doctest',
+    'sphinx.ext.extlinks',
+    'sphinx.ext.intersphinx',
+    'sphinx.ext.todo',
+    'sphinx.ext.coverage',
+    'sphinx.ext.viewcode',
+
+    # Third-party extensions:
+    'myst_parser',  # extended markdown; https://pypi.org/project/myst-parser/
+    'sphinxcontrib.apidoc',
+]
+
+# Conditional third-party extensions:
+try:
+    import sphinxcontrib.spelling as _sphinxcontrib_spelling
+except ImportError:
+    extensions.append('spelling_stub_ext')
+else:
+    del _sphinxcontrib_spelling
+    extensions.append('sphinxcontrib.spelling')
+
+# The language for content autogenerated by Sphinx. Refer to documentation
+# for a list of supported languages.
+#
+# This is also used if you do content translation via gettext catalogs.
+# Usually you set "language" from the command line for these cases.
+language = 'en'
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+html_theme = 'furo'
+
+html_show_sphinx = True
+
+html_theme_options = {
+}
+
+html_context = {
+}
+
+
+# The name for this set of Sphinx documents.  If None, it defaults to
+# "<project> v<release> documentation".
+# html_title = f'{project} Documentation'
+
+# A shorter title for the navigation bar.  Default is the same as html_title.
+# html_short_title = 'Documentation'
+
+# If not '', a 'Last updated on:' timestamp is inserted at every page bottom,
+# using the given strftime format.
+html_last_updated_fmt = '%b %d, %Y'
+
+# If true, an OpenSearch description file will be output, and all pages will
+# contain a <link> tag referring to it.  The value of this option must be the
+# base URL from which the finished HTML is served.
+# html_use_opensearch = f'https://{github_repo_name.replace(".", "")}.readthedocs.io/en/latest/'
+
+# The master toctree document.
+root_doc = master_doc = 'index'  # Sphinx 4+ / 3-  # noqa: WPS429
+
+
+# -- Extension configuration -------------------------------------------------
+
+# -- Options for intersphinx extension ---------------------------------------
+
+intersphinx_mapping = {
```

Comment:
@abhinavsingh intersphinx allows you to reference objects defined on other Sphinx sites for as long as you list them here. Sphinx will download their `objects.inv` files with lists of everything exposed.
The built-in way of exploring those objects is:
```console
$ python -m sphinx.ext.intersphinx https://docs.python.org/3/objects.inv
```

But it's not very user-friendly, so I have a small project displaying this information for several sites I sometimes interact with. It's here: https://webknjaz.github.io/intersphinx-untangled/.

When linking objects from a foreign site, you can type them in as per usual, with their correct role and name. But since objects with the same name may exist on this and other sites, I prefer explicitly predending the references with `<site-alias>:`. The dict keys below are those aliases you can use to point at objects at a specific site.